bakery/codec_test.go

```diff
@@ -24,127 +26,148 @@ func (s *codecSuite) SetUpTest(c *gc.C) {
 }
 
 func (s *codecSuite) TestV1RoundTrip(c *gc.C) {
-	cid, err := encodeCaveatIdV1(
+	cid, err := encodeCaveatV1(
 		"is-authenticated-user", []byte("a random string"), &s.thirdPartyKey.Public, s.firstPartyKey)
 
 	c.Assert(err, gc.IsNil)
 
-	res, err := decodeCaveatId(s.thirdPartyKey, cid)
+	res, err := decodeCaveat(s.thirdPartyKey, cid)
 	c.Assert(err, gc.IsNil)
 	c.Assert(res, jc.DeepEquals, &ThirdPartyCaveatInfo{
 		FirstPartyPublicKey: s.firstPartyKey.Public,
 		RootKey:             []byte("a random string"),
 		Condition:           "is-authenticated-user",
-		CaveatId:            cid,
-		MacaroonId:          cid,
+		Caveat:              cid,
 		ThirdPartyKeyPair:   *s.thirdPartyKey,
 		Version:             Version1,
+		Namespace:           legacyNamespace(),
 	})
 }
 
 func (s *codecSuite) TestV2RoundTrip(c *gc.C) {
-	cid, err := encodeCaveatIdV2("is-authenticated-user", []byte("a random string"), &s.thirdPartyKey.Public, s.firstPartyKey)
+	cid, err := encodeCaveatV2("is-authenticated-user", []byte("a random string"), &s.thirdPartyKey.Public, s.firstPartyKey)
 
 	c.Assert(err, gc.IsNil)
 
-	res, err := decodeCaveatId(s.thirdPartyKey, cid)
+	res, err := decodeCaveat(s.thirdPartyKey, cid)
 	c.Assert(err, gc.IsNil)
 	c.Assert(res, jc.DeepEquals, &ThirdPartyCaveatInfo{
 		FirstPartyPublicKey: s.firstPartyKey.Public,
 		RootKey:             []byte("a random string"),
 		Condition:           "is-authenticated-user",
-		CaveatId:            cid,
-		MacaroonId:          cid,
+		Caveat:              cid,
 		ThirdPartyKeyPair:   *s.thirdPartyKey,
 		Version:             Version2,
+		Namespace:           legacyNamespace(),
+	})
+}
+
+func (s *codecSuite) TestV3RoundTrip(c *gc.C) {
+	ns := checkers.NewNamespace(nil)
+	ns.Register("testms", "x")
```

Comment:
I suspect you mean "testns" I had to think for a while to try and work out what testms might be.